sharding/nodesSetup_test.go

```diff
@@ -514,3 +726,340 @@ func TestNodesSetup_NodesAboveMaxShardLimit(t *testing.T) {
+
+	for i := 0; i < 2; i++ {
+		convertedValue, err := pubKeyConverter.Decode(listOfInitialNodes[i].PubKey)
+		require.Nil(t, err)
+		require.NotNil(t, convertedValue)
+		listOfExpectedPubKeysEligibleNodes[i] = string(convertedValue)
+	}
+
+	ns, err := createTestNodesSetup(argsTestNodesSetup{
+		shardConsensusSize: 1,
+		shardMinNodes:      1,
+		metaConsensusSize:  1,
+		metaMinNodes:       1,
+		numInitialNodes:    3,
+		genesisMaxShards:   1,
+	})
+	require.Nil(t, err)
+	ns.Hysteresis = 0.2
+	ns.Adaptivity = false
+
+	ns = createAndAssignNodes(ns, noOfInitialNodes)
+
+	allEligibleNodes, err := ns.InitialEligibleNodesPubKeysForShard(0)
+
+	require.Nil(t, err)
+	require.Equal(t, listOfExpectedPubKeysEligibleNodes[1], allEligibleNodes[0])
+}
+
+func TestNodesSetup_NumberOfShardsShouldWork(t *testing.T) {
+	t.Parallel()
+
+	noOfInitialNodes := 3
+
+	ns, err := createTestNodesSetup(argsTestNodesSetup{
+		shardConsensusSize: 1,
+		shardMinNodes:      1,
+		metaConsensusSize:  1,
+		metaMinNodes:       1,
+		numInitialNodes:    3,
+		genesisMaxShards:   1,
+	})
+	require.Nil(t, err)
+	require.NotNil(t, ns)
+
+	ns.Hysteresis = 0.2
+	ns.Adaptivity = false
+
+	ns = createAndAssignNodes(ns, noOfInitialNodes)
+
+	require.NotNil(t, ns)
+
+	valReturned := ns.NumberOfShards()
+	require.Equal(t, uint32(1), valReturned)
+
+	valReturned = ns.MinNumberOfNodesWithHysteresis()
+	require.Equal(t, uint32(2), valReturned)
+
+	valReturned = ns.MinNumberOfShardNodes()
+	require.Equal(t, uint32(1), valReturned)
+
+	valReturned = ns.MinNumberOfShardNodes()
+	require.Equal(t, uint32(1), valReturned)
+
+	shardConsensusGroupSize := ns.GetShardConsensusGroupSize()
+	require.Equal(t, uint32(1), shardConsensusGroupSize)
+
+	metaConsensusGroupSize := ns.GetMetaConsensusGroupSize()
+	require.Equal(t, uint32(1), metaConsensusGroupSize)
+
+	ns.Hysteresis = 0.5
+	hysteresis := ns.GetHysteresis()
+	require.Equal(t, float32(0.5), hysteresis)
+
+	ns.Adaptivity = true
+	adaptivity := ns.GetAdaptivity()
+	require.True(t, adaptivity)
+
+	ns.StartTime = 2
+	startTime := ns.GetStartTime()
+	require.Equal(t, int64(2), startTime)
+
+	ns.RoundDuration = 2
+	roundDuration := ns.GetRoundDuration()
+	require.Equal(t, uint64(2), roundDuration)
+
+}
+
+func TestNodesSetup_ExportNodesConfigShouldWork(t *testing.T) {
+	t.Parallel()
+
+	noOfInitialNodes := 3
+
+	ns, err := createTestNodesSetup(argsTestNodesSetup{
+		shardConsensusSize: 1,
+		shardMinNodes:      1,
+		metaConsensusSize:  1,
+		metaMinNodes:       1,
+		numInitialNodes:    3,
+		genesisMaxShards:   1,
+	})
+	require.Nil(t, err)
+
+	ns.Hysteresis = 0.2
+	ns.Adaptivity = false
+	ns.StartTime = 10
+
+	ns = createAndAssignNodes(ns, noOfInitialNodes)
+	configNodes := ns.ExportNodesConfig()
+
+	require.Equal(t, int64(10), configNodes.StartTime)
+
+	for i, nodeConfig := range configNodes.InitialNodes {
+		require.Equal(t, config.InitialNodeConfig{PubKey: pubKeys[i], Address: address[i], InitialRating: 0},
```

Comment:
can we extract the `expectedNodesConfig` and `actualNodesConfig` into variables and then have the require much easier to read?